src/main.rs

```diff
@@ -10,51 +10,66 @@ mod helpers;
 mod protocol;
 mod server;
 
-#[derive(Clone, ValueEnum, Debug)]
-enum Role {
-    Client,
-    Server,
-}
-
-#[derive(Clone, ValueEnum, Debug)]
-pub enum GraphType {
-    Mermaid,
-    Dot,
-}
-
+/// A simple echo server & client generated using the Hydroflow template. The lines starting with
+/// `///` contain the message that appears when you run the compiled binary with the '--help'
+/// arguments, so feel free to change it to whatever makes sense for your application.
 #[derive(Parser, Debug)]
 struct Opts {
+
+    // The `Opts` structure contains the command line arguments accepted by the application and can
+    // be modified to suit your requirements. Refer to the clap crate documentation for more
+    // information.
+
+    /// The role this application process should assume. The example in the template provides two
+    /// roles: server and client. The server echoes whatever message the clients send to it.
+    #[clap(value_enum, long)] // value_enum => parse as enum. long => "--role" instead of "-r".
+    role: Role, // This is a mandatory argument.
+
+    /// The server's network address. The server listens on this address. The client sends messages
+    /// to this address.
+    #[clap(long, value_parser = ipv4_resolve)] // value_parser => parse "ip:port" using ipv4_resolve
+    address: Option<SocketAddr>, // Since this is an Option<T>, it is an optional argument.
+
+    /// If specified, a graph representation of the Hydroflow flow used by the program will be
+    /// printed to the console in the specified format. This parameter can be removed if your
+    /// application doesn't need this functionality.
     #[clap(value_enum, long)]
-    role: Role,
-    // #[clap(long)]
-    #[clap(long, value_parser = ipv4_resolve)]
-    addr: Option<SocketAddr>,
-    // #[clap(long)]
-    #[clap(long, value_parser = ipv4_resolve)]
-    server_addr: Option<SocketAddr>,
-    #[clap(value_enum, long)]
-    graph: Option<GraphType>,
+    graph: Option<GraphType>, // This too, is an optional argument.
```

Comment:
Actually, would be good to update this to use the args from `use hydroflow::lang::graph::{WriteConfig, WriteGraphType};`

```rust
    #[clap(long)]
    graph: Option<WriteGraphType>,
    #[clap(flatten)]
    write_config: Option<WriteConfig>,
```

Example usage:

https://github.com/hydro-project/hydroflow/blob/550b17c0e0edd233b0cceab55fed0de309051503/hydroflow/examples/kvs/server.rs#L36-L41

Reply:
Done!